examples/handle-head-request/src/main.rs

```diff
@@ -0,0 +1,95 @@
+//! Run with
+//!
+//! ```not_rust
+//! cd examples && cargo run -p example-handle-head-reqeust
+//! ```
+
+use axum::response::{IntoResponse, Response};
+use axum::{http, routing::get, Router};
+use std::net::SocketAddr;
+
+fn app() -> Router {
+    Router::new().route("/get-head", get(get_head_handler))
+}
+
+#[tokio::main]
+async fn main() {
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+
+    axum::Server::bind(&addr)
+        .serve(app().into_make_service())
+        .await
+        .unwrap();
+}
+
+// get routes will also be called for HEAD requests but will have the response body removed
+// can handle implicit head method request by extract `http::Method` from request
+async fn get_head_handler(method: http::Method) -> Response {
+    let something = Some(Thing);
+
+    // it would usually handle only special-case HEAD
+    // if computing the body has some relevant compute cost
```

Comment:
Double 'compute' sounds a bit weird

```suggestion
    // if computing the body has some relevant cost
```